compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2820,6 +2820,21 @@ impl<'a> Parser<'a> {
                 }
                 self.recover_struct_comma_after_dotdot(exp_span);
                 break;
+            } else if self.token == token::DotDotDot {
+                // suggest `..v` instead of `...v`
+                let snapshot = self.create_snapshot_for_diagnostic();
+                let span = self.token.span;
+                self.bump();
+                match self.parse_expr() {
+                    Ok(_p) => {
+                        self.sess.emit_err(MissingDotDot { token_span: span, sugg_span: span });
```

Comment:
Looking better! 

Can you actually set the base expression like above? Something like `base = ast::StructRest::Base(e)`. Basically emulate the code above. That should suppress the "missing fields `x` and `y` in initializer of `V3`" error below.

Also, can handle the case of `... }` like above?

Reply:
oo, I left the "missing fields x and y in initializer of V3" intentionally, I thought it may helpful for let developer know which fields are missing.

And for the case of `...}`, we need to suggest something like `..{expr}`? So I left it as before also.

Reply:
> I thought it may helpful for let developer know which fields are missing.

This is extra noise if the developer wrote `...` instead of `..`

> And for the case of `...}`, we need to suggest something like `..{expr}`?

Not in the case of a pattern, which is the same place that `.. }` is valid.

---

Basically, we should treat `...` typo as closely to `..` as possible, especially if we're ¨recovering" parsing here.

Reply:
It may actually be easier to merge this check with the `..` logic above, since we're basically just duplicating that.

Reply:
Ok, I will make a update.

Reply:
For the case of `let V3 { z: val, ...} = v`, we already have a check to report:

```console
error: expected field pattern, found `...`
  --> /home/cat/code/rust/src/test/ui/parser/issue-102806.rs:23:22
   |
23 |     let V3 { z: val, ... } = v;
   |                      ^^^ help: to omit remaining fields, use one fewer `.`: `..`
```

Will keep same as before.

Reply:
> It may actually be easier to merge this check with the `..` logic above, since we're basically just duplicating that.

Tried to merge the two checks, but I found split them seems more easy to read.

Reply:
That's ok, I don't think it's necessary

Reply:
Finally, merge it with the `..` logic above 😁